contracts/colony/Colony.sol

```diff
@@ -49,25 +49,34 @@ contract Colony is ColonyStorage, PatriciaTreeProofs, MultiChain {
   public stoppable auth
   returns (bool)
   {
-    return makeArbitraryTransactionFunctionality(_to, _action);
+    return this.makeSingleArbitraryTransaction(_to, _action);
   }
 
-  function makeArbitraryTransactions(address[] memory _targets, bytes[] memory _actions)
+  function makeArbitraryTransactions(address[] memory _targets, bytes[] memory _actions, bool _strict)
   public stoppable auth
   returns (bool)
   {
     require(_targets.length == _actions.length, "colony-targets-and-actions-length-mismatch");
-    for (uint256 i; i < _targets.length; i+= 1){
-      require(
-        makeArbitraryTransactionFunctionality(_targets[i], _actions[i]),
-        "colony-arbitrary-transaction-failed"
-      );
+    for (uint256 i; i < _targets.length; i += 1){
+      bool success = true;
+      try this.makeSingleArbitraryTransaction(_targets[i], _actions[i]) returns (bool ret){
```

Comment:
I would have done this differently:

```
bool success = true;
for (uint256 i; i < _targets.length; i += 1) {
   success = success && this.makeSingleArbitraryTransaction(_targets[i], _actions[i]);
   require(success || !_strict, "colony-arbitrary-transaction-failed);
}
return success;

Reply:
I believe we have to use try and an external call here because of the `require`s in `makeSingleArbitraryTransaction` - if one of those fail, with your pattern, the whole transaction fails even if `_strict` is false.

Reply:
Also why `this`? We had it as an internal call earlier it it was all groovy.

Reply:
> I don't follow -- makeSingleArbitraryTransaction returns a boolean, doesn't it?

If a require fails, it does not return a boolean and instead the whole transactions fails.

> It doesn't have an internal require.

The first three lines in the function are `require`s, at least? There are then also `require`s in the `preparation` functions.

> Also why this? We had it as an internal call earlier it it was all groovy.

`try` only works on external calls

Reply:
Oh, right -- I was thinking specifically of the `executeCall` but there are a lot of validation checks which could throw. All good!